### ISSUE TITLE
crash after going back through nether portal to overworld

### ISSUE THREAD

**Issue by ymnejmi** [MCRcortex/nvidium#106]
as mentioned in title, i tried multiple worlds with no difference
[crash-2023-11-21_09.30.30-client.txt](https://github.com/MCRcortex/nvidium/files/13450445/crash-2023-11-21_09.30.30-client.txt)
gpu: gtx 1650
os: linux/xwayland
client: prismlauncher 
version: 1.20.2
if more information/tests are need ill be happy to assist

**Comment by flipfloppy1** [comment]
I’ve had this issue too, it seems to happen sporadically

**Comment by ichijiyujin** [comment]
I'm seeing the same thing, also Linux. I suspect a memory leak might be involved. I noticed that when going through a nether portal the RAM usage of Minecraft jumped from ~4GB to up to 11GB (when also using other mods, and only jumps to ~7GB when I use only Sodium and Nvidium). If I'm lucky, it won't crash at the first pass through a portal, but the memory usage remains high and it's guaranteed to crash the next time simply because it ends up using all my available memory. I also noticed that repeatedly disabling and reenabling Nvidium from within the game increases the memory usage in a similar way, though in only 2GB increments each time I reenable nvidium.

**Comment by flipfloppy1** [comment]
Also happening on windows for me. It seems like an issue with releasing memory after the renderer has unloaded.

**Comment by Holtigo** [comment]
Bump,
This is still an issue in the current version, from what I can see. Going from any dimension back to the overworld, most of the time, will start to make the game stutter heavily, after which it will crash most of the time. Maybe an issue unloading the chunks from whichever dimension when returning to overworld, either way, this issue still persists.

**Comment by MCRcortex** [comment]
Same issue as #165 please test without any other mods

**Comment by ymnejmi** [comment]
My setup now is pretty now from what I had back then and I don't play much minecraft lately, but I tried flying around in spectator mode a while in the nether and back a few times and the memory looks stable and no longer jumps up every time (stable 40% as opposed to 80+% and crash).

I'm soo happy that even my +5 year old laptop can run minecraft at 64 chunks, its really crazy. Thank you so much for your work I dont even understand how it works it feels like magic :smile: 

tldr; looks fixed to me so I'll close it.

**Comment by MCRcortex** [comment]
<3 thank you very much for going back and checking!
